glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt

```diff
@@ -24,20 +24,25 @@ enum class UploadTaskTag {
     Done
 }
 
-@Structure.FieldOrder("tag", "documentId", "path", "body", "headers")
+@Structure.FieldOrder("tag", "documentId", "path", "bodyLen", "body", "headers")
 internal class UploadBody(
     // NOTE: We need to provide defaults here, so that JNA can create this object.
     @JvmField val tag: Byte = UploadTaskTag.Done.ordinal.toByte(),
     @JvmField val documentId: Pointer? = null,
     @JvmField val path: Pointer? = null,
-    @JvmField val body: Pointer? = null,
+    // Note that the next two fields (`bodyLen` and `body`) are defined as a single
+    // structure, on the Rust side, called `ByteBuffer`. While the ideal would be to
```

Comment:
Have you used the [`RustBuffer`](https://github.com/mozilla/application-services/blob/e3670c45716b496b550178cc5ed48fbdede816d3/components/support/android/src/main/java/mozilla/appservices/support/native/RustBuffer.kt#L44) structure from a-s?
You might also define this here as `body: RustBuffer.ByValue`

Reply:
Yes, that's the first thing I tried, but still the same problem: when using `union` it fails because it can't determine the size of the struct.

Reply:
Hm, that's ... weird and unfortunate.

Reply:
Can we file a bug so we can look into this later again?

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1637611